lib/ansible/modules/packaging/os/rhsm_repository.py

```diff
@@ -0,0 +1,265 @@
+#!/usr/bin/python
+
+# Giovanni Sciortino (@giovannisciortino)
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: rhsm_repository
+short_description: Manage RHSM repositories using the C(subscription-manager)
+                   command
+description:
+  - Manage(List/Enable/Disable) RHSM repositories to the Red Hat Subscription
+    Management entitlement platform using the C(subscription-manager) command.
+version_added: '2.5'
+author: Giovanni Sciortino (@giovannisciortino)
+notes:
+  - In order to manage RHSM repositories the system must be already registered
+    to RHSM manually or using the ansible module redhat_subscription.
+  - One option between name and list must be defined, both options in the
+    same task must not be defined.
+
+requirements:
+  - subscription-manager
+options:
+  state:
+    description:
+      - If state is equal to enabled or disabled, indicates the desired
+        repository state.
+      - If state is equal to list, list_enabled, or list_disabled, list
+        C(all)/C(enabled)/C(disabled) repositories.
+    choices: [enabled, disabled, list, list_enabled, list_disabled]
+    required: False
```

Comment:
If it is not required, you don't have to state it as it's the default.

Reply:
state option is required and it has the default option value "enabled". The code was already correct, I have fixed the documentation.